sdk/storage/Azure.Storage.Blobs.Batch/src/BlobBatchClient.cs

```diff
@@ -267,61 +340,95 @@ private class BlobServiceClientInternals : BlobServiceClient
             bool async,
             CancellationToken cancellationToken)
         {
-            batch = batch ?? throw new ArgumentNullException(nameof(batch));
-            if (batch.Submitted)
-            {
-                throw BatchErrors.CannotResubmitBatch(nameof(batch));
-            }
-            else if (!batch.IsAssociatedClient(this))
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(BlobBatchClient)}.{nameof(SubmitBatch)}");
+            try
             {
-                throw BatchErrors.BatchClientDoesNotMatch(nameof(batch));
-            }
+                scope.Start();
 
-            // Get the sub-operation messages to submit
-            IList<HttpMessage> messages = batch.GetMessagesToSubmit();
-            if (messages.Count == 0)
-            {
-                throw BatchErrors.CannotSubmitEmptyBatch(nameof(batch));
-            }
-            // TODO: Consider validating the upper limit of 256 messages
+                batch = batch ?? throw new ArgumentNullException(nameof(batch));
+                if (batch.Submitted)
+                {
+                    throw BatchErrors.CannotResubmitBatch(nameof(batch));
+                }
+                else if (!batch.IsAssociatedClient(this))
+                {
+                    throw BatchErrors.BatchClientDoesNotMatch(nameof(batch));
+                }
```

Comment:
Maybe instead of this we should have `blobBatchClient.CreateBatch().Submit()`. I.e. move `Submit` to `BlobBatch` and hide Submit on BlobBatchClient. BlobBatch can hold reference to client it originated from.

Current shape of API indicates that I can hand BlobBatch to any BlobBatchClient. Which isn't reality here.

Reply:
I believe this functionality was already here - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs.Batch/src/BlobBatchClient.cs#L275